File: 数组枚举/数组枚举/ViewController.m
//
//  ViewController.m
//  数组枚举
//
//  Created by BaiLun on 2018/11/14.
//  Copyright © 2018 bailun. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (nonatomic, strong) NSMutableArray *items;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _items = [NSMutableArray array];
    [self insertObject:100000];
   
}

- (void)insertObject:(NSInteger)count {
    for (NSInteger i = 0; i < count; i++) {
        [_items addObject:@(i)];
    }
}

- (void)enumItems {
    NSArray *array = self.items;
    for (NSNumber *num in array) {
        NSLog(@"testdata----%ld", num.integerValue);
    }
}

- (IBAction)handleArrayEnum:(id)sender {
    
    [self enumItems];
//    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
//        [self insertObject:1000];
//    });
}

@end

File: BLVideoDemo/BLVideoDemo/ViewController.swift
//
//  ViewController.swift
//  BLVideoDemo
//
//  Created by BaiLun on 2018/10/13.
//  Copyright © 2018 qinrongjun. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var url: URL?

    @IBOutlet weak var sliderBar: UISlider!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        SessionDrafPlistManager.shared.createPlist(name: "test.plist", removeOld: true)
        for item in 0..<10 {
            SessionDrafPlistManager.shared.addValue(to: "test.plist", value: arc4random()/10000, key: String(describing: item))
        }
        sliderBar.setThumbImage(UIImage(named: "dot"), for: .normal)
        sliderBar.setThumbImage(UIImage(named: "dot"), for: .highlighted)
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        self.play()
    }
    
    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
        self.url = nil
    }

    @IBAction func slider(_ sender: UISlider) {
        print(sender.value)
    }
    
    @IBAction func startRecord(_ sender: UIButton) {
        
        let capatureViewController = CaptureViewController()
        capatureViewController.delegate  = self
        self.present(capatureViewController, animated: true, completion: nil)
    }
    @IBAction func add(_ sender: Any) {
        SessionDrafPlistManager.shared.addValue(to: "test.plist", value: "9999", key: "100")
    }

    @IBAction func deleteItem(_ sender: Any) {
        SessionDrafPlistManager.shared.removeValue(from: "test.plist", key: "100")
    }
    
    @IBAction func query(_ sender: Any) {
        print(SessionDrafPlistManager.shared.queryValue(from: "test.plist", key: "1") ?? "未找到这个值")
    }
    
    @IBAction func update(_ sender: Any) {
        SessionDrafPlistManager.shared.updateValue(to: "test.plist", newValue: "ppdfsjiaf", key: "1")
    }
    
    @IBAction func showAllItems(_ sender: Any) {
      
    }
    
    @IBAction func playVideo(_ sender: Any) {
        if  let urlStr = Bundle.main.path(forResource: "kongfu", ofType: "mp4") {
            url = URL(fileURLWithPath: urlStr)
            self.play()
        }
    }
    
    func play() {
        if let videoURL  = self.url {
            let playerModel = PlayerModel()
            playerModel.videoURL = videoURL
            let player = KFCPlayer(model: playerModel)
            player.show(from: self.view)
        }
    }
}

extension ViewController: CaptureViewControllerDelegate {
    func captureViewController(vc: CaptureViewController, didFinishTakePhoto photoData: Data?, error: Error?) {
        
    }
    
    func captureViewController(vc: CaptureViewController, didFinishRecordVideo videoURL: URL, error: Error?) {
       self.url = videoURL
    }
    
   
}

